### PR TITLE
Add test for false table reference negative

### PR DESCRIPTION
**Pull request by crisptrutski**
This is an example where a query reads from two tables, but the analyzer claims that it read from neither.

### CODE REVIEW

**Review by tsmacdonald — APPROVED** [review]
:(

File: test/resources/acceptance/compound__cte_masking.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+WITH c AS (SELECT x FROM b),
+     b AS (SELECT y FROM a),
+     a AS (SELECT x FROM c)
```
Comment (crisptrutski) [inline test/resources/acceptance/compound__cte_masking.sql:3]: Worth noting that Postgres rejects this query, thinking that is has forward references. MySQL happily executes it and returns the product of a and b.
